frontend/src/instructorView/instructorView.js

```diff
@@ -0,0 +1,378 @@
+                            <div className="row">
+                                {
+                                    Object.keys(modules).map((module, k) => (
+                                        <Module
+                                            key={k}
+                                            moduleName={module}
+                                            moduleInfo={modules[module]}
+                                            updateModule={this.updateModule}
+                                        />
+                                    ))
+                                }
+                            </div>
+                        </div>
+                    </div>
+                    <div className="card-button-div">
+                        <button
+                            onClick={this.saveText}
+                            className={`card-buttons save-button ${editing ? 'disabled' : ''}`}
+                            disabled={editing}
+                        >
+                            Save
+                        </button>
+                        <button
+                            onClick={this.deleteText}
+                            className={`card-buttons delete-button ${editing ? 'disabled' : ''}`}
+                            disabled={editing}
+                        >
+                            Delete
+                        </button>
+                    </div>
+                </div>
+            </div>
+        );
+    }
+}
+TextInfo.propTypes = {
+    text: PropTypes.object,
+    delete: PropTypes.func,
+};
+
+export class InstructorView extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            textData: null,
+            showModal: false,
+            addingText: 0,
+            addTitle: '',
+            addContent: '',
+        };
+        this.modalHandler = this.modalHandler.bind(this);
+    }
+
+    async componentDidMount() {
+        try {
+            const apiURL = '/api/all_text';
+            const response = await fetch(apiURL);
+            const data = await response.json();
+            this.setState({
+                textData: data,
+            });
+        } catch (e) {
+            console.log(e);
+        }
+    }
+
+    deleteText = (id) => {
+        let { textData } = this.state;
+        textData = textData.filter((text) => text.id !== id);
+        this.setState({ textData });
+    }
+
+    modalHandler = (event) => {
+        event.preventDefault();
+        this.setState({
+            showModal: !this.state.showModal,
+        });
+    }
+
+    handleInput = (event) => {
+        if (event.target.id === 'title') {
+            this.setState({
+                addTitle: event.target.value,
+            });
+        } else {
+            this.setState({
+                addContent: event.target.value,
+            });
+        }
+    }
+
+    handleSubmit = async (event) => {
```

Comment:
This looks good: but let's think about how we can structure this moving forward so that the instructor doesn't need to leave thee window open while the text is processed on our end.